src/vs/workbench/parts/tasks/common/problemMatcher.ts

```diff
@@ -428,8 +434,8 @@ class MultiLineMatcher extends AbstractLineMatcher {
 	public handle(lines: string[], start: number = 0): HandleResult {
 		Assert.ok(lines.length - start === this.patterns.length);
 		this.data = Object.create(null);
-		let data = this.data;
-		data.kind = this.patterns[0].kind;
+		let data = this.data!;
```

Comment:
And here neither. I see that the TS compile still complains so I would suggest you file an
issue against the TS repository. I don't see why they can't esnure that this.data looses null 
in this case